docs/src/main/asciidoc/using-vertx.adoc

```diff
@@ -0,0 +1,483 @@
+They are just shims and rely on a single _managed_ Vert.x instance.
+
+You will pick one or the other depending on your use cases.
+
+- `bare`: for advanced usage or if you have existing Vert.x code you want to reuse in your {project-name} application
+- `Axle`: works well with {project-name} and MicroProfile APIs (`CompletionStage` for single results and `Publisher` for streams)
+- `Rx Java 2`: when you need support for a wide range of data transformation operators on your streams
+
+The following snippets illustrate the difference between these 3 APIs:
+
+[source, java]
+----
+// Bare Vert.x:
+vertx.fileSystem().readFile("lorem-ipsum.txt", ar -> {
+    if (ar.succeeded()) {
+        System.out.println("Content:" + ar.result().toString("UTF-8"));
+    } else {
+        System.out.println("Cannot read the file: " + ar.cause().getMessage());
+    }
+});
+
+// Rx Java 2 Vert.x
+vertx.fileSystem().rxReadFile("lorem-ipsum.txt")
+    .map(buffer -> buffer.toString("UTF-8"))
+    .subscribe(
+            content -> System.out.println("Content: " + content),
+            err -> System.out.println("Cannot read the file: " + err.getMessage())
+    );
+
+// Axle API:
+vertx.fileSystem().readFile("lorem-ipsum.txt")
+    .thenApply(buffer -> "Content: " + buffer.toString("UTF-8"))
+    .exceptionally(err -> "Cannot read the file: " + err.getMessage())
+    .thenAccept(s -> System.out.println(s));
+----
+
+== Using Vert.x in Reactive Jax-RS resources
+
+{project-name} web resources support asynchronous processing and streaming results over https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events[server-sent events].
+
+=== Asynchronous processing
+
+Most programming guides start easy with a greeting service and this one makes no exception.
+
+To asynchronously greet a client, the endpoint method must return a `java.util.concurrent.CompletionStage`:
+
+[source,java]
+----
+@Path("/hello")
+public class GreetingResource {
+
+    @Inject
+    Vertx vertx;
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    @Path("{name}")
+    public CompletionStage<String> greeting(@PathParam("name") String name) {
+        // When complete, return the content to the client
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        long start = System.nanoTime();
+
+        // TODO: asynchronous greeting
+
+        return future;
+    }
+}
+----
+
+So far so good.
+Now let's use the Vert.x API to implement the artificially delayed reply with the `setTimer` provided by Vert.x:
+
+[source,java]
+----
+// Delay reply by 10ms
+vertx.setTimer(10, l -> {
+    // Compute elapsed time in milliseconds
+    long duration = MILLISECONDS.convert(System.nanoTime() - start, NANOSECONDS);
+
+    // Format message
+    String message = String.format("Hello %s! (%d ms)%n", name, duration);
+
+    // Complete
+    future.complete(message);
```

Comment:
Doesn't the vert axle flavour have a version of `setTimer` that returns a `CompletionStage` we can use to simplify this example?

Reply:
That's a good idea, but no, it does not provide this method at the moment. 
The only difference with the bare vert.x version is the usage of `Consumer` instead of `Handler`